lib/utils/process-jobs.js

```diff
@@ -285,9 +285,12 @@ module.exports = function(extraJob) {
 
           // CALL THE ACTUAL METHOD TO PROCESS THE JOB!!!
           debug('[%s:%s] processing job', job.attrs.name, job.attrs._id);
-          job.run()
-            .catch(err => [err, job])
-            .then(job => processJobResult(...Array.isArray(job) ? job : [null, job]));
+          try {
+            const runJob = await job.run();
+            processJobResult(...Array.isArray(runJob) ? runJob : [null, runJob]);
+          } catch (err) {
+            throw err;
```

Comment:
This is not the same behaviour.

```javascript
try {
    processJobResult(null, await job.run());
  } catch (err) {
    processJobResult(err);
  }
```

is probably closer, but then you potentially call `processJobResult` again if it is what throws the error 🤔